tools/box/box.h

```diff
@@ -53,38 +56,122 @@ jxl::Status ParseBoxHeader(const uint8_t** next_in, size_t* available_in,
+  To facilitate transparent handling of boxes, the BrobBlob struct is used.
+  It has pointers to both the uncompressed data and the compressed data.
+  When decoding:
+    - when reading, either udata* or cdata* will be set
+    - in the uncompressed case, only udata* is used.
+    - in the compressed case, cdata* points into the input stream, and
+      when accessing the box with getBlob(), pbytes will store the
+      decompressed data, and udata* will point to it.
+  When encoding:
+    - in the uncompressed case, only udata* is used.
+    - in the compressed case, udata* points to the uncompressed input, and
+      when adding the box with addBlob(), pbytes will store the compressed
+      data, and cdata* will point to it.
+    - when writing, a "brob" box will be written if cdata* is nonzero,
+      otherwise an uncompressed box will be written.
+*/
+struct BrobBlob {
+  uint8_t type[4];
+  // uncompressed data (mutable: decompressed data can be added or removed)
+  mutable const uint8_t* udata = nullptr;  // Not owned
+  mutable size_t udata_size = 0;
+  // compressed data
+  const uint8_t* cdata = nullptr;  // Not owned
+  size_t cdata_size = 0;
+  // a buffer to store the (de)compressed data
+  mutable jxl::PaddedBytes pbytes;  // Owned
+};
+
 // NOTE: after DecodeJpegXlContainerOneShot, the exif etc. pointers point to
 // regions within the input data passed to that function.
 struct JpegXlContainer {
-  // Exif metadata, or null if not present in the container.
-  // The exif data has the format of 'Exif block' as defined in
-  // ISO/IEC23008-12:2017 Clause A.2.1
-  // Here we assume the tiff header offset is 0 and store only the
-  // actual Exif data (starting with the tiff header MM or II)
-  // TODO(lode): support the theoretical case of multiple exif boxes
-  const uint8_t* exif = nullptr;  // Not owned
-  size_t exif_size = 0;
-
-  // Brotli-compressed exif metadata, if present. The data points to the brotli
-  // compressed stream, it is not decompressed here.
-  const uint8_t* exfc = nullptr;  // Not owned
-  size_t exfc_size = 0;
-
-  // XML boxes for XMP. There may be multiple XML boxes.
-  // Each entry points to XML location and provides size.
-  // The memory is not owned.
-  // TODO(lode): for C API, cannot use std::vector.
-  std::vector<std::pair<const uint8_t*, size_t>> xml;
-
-  // Brotli-compressed xml boxes. The bytes are given in brotli-compressed form
-  // and are not decompressed here.
-  std::vector<std::pair<const uint8_t*, size_t>> xmlc;
-
-  // JUMBF superbox data, or null if not present in the container.
-  // The parsing of the nested boxes inside is not handled here.
-  const uint8_t* jumb = nullptr;  // Not owned
-  size_t jumb_size = 0;
-
+  std::vector<BrobBlob> blobs;
+
+  // Returns a pointer to a BrobBlob with uncompressed data (or nullptr if not
+  // found). In case there are multiple of that type, return the index-th one.
+  const BrobBlob* getBlob(const char* type, size_t index = 0) const {
```

Comment:
Is there a reason to not have this in the .cc file instead? These should be CamelCase method names.